<?xml version="1.0"?>
<launch>

  <arg name="robot_model"            default="hera_full"/>
  <arg name="robot_name"             default="robot"/>
  

  <arg name="enable_rviz"            default="true"/>
  <arg name="enable_gazebo"          default="true"/>
  <arg name="enable_teleop"          default="true"/>
  <arg name="enable_check_collision" default="false"/>

  <!-- Task evaluator args -->
  <arg name="task_evaluator" default="false" />
  <arg name="camera_controller" default="true" />
  <!-- Randomization args -->
  <arg name="spawn_objects" default="true"/>
  <arg name="seed" default="1" />
  <arg name="per_category" default="6" />
  <arg name="obstacles" default="4" />
  <arg name="per_row" default="6" />
  
  <param name="/use_sim_time" value="true"/>
  <param name="gazebo/enable_ros_network" value="true" />

  <!-- use rviz -->
  <group if="$(arg enable_rviz)">
    <node name="rviz" pkg="rviz" type="rviz" output="screen"
      args="-d $(find hera_description)/config/rviz/vizualize_model.rviz"/>
  </group>

  <!-- use gazebo -->
  <env name="GAZEBO_MODEL_PATH" value="$GAZEBO_MODEL_PATH:$(find social_worlds)/models/:$(find social_worlds)/models/3dparty/"/>
  <group if="$(arg enable_gazebo)">
    <!-- gazebo -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch"/>
    <!-- <include file="$(find tmc_wrs_gazebo_worlds)/worlds/wrs2020.world"/> -->
    <!-- load hera description -->
    <include file="$(find hera_description)/launch/load_description.launch">
      <arg name="robot_model" default="$(arg robot_model)"/>
      <arg name="robot_name" 	default="$(arg robot_name)"/>
      <arg name="use_jsp_gui" default="false"/>
    </include>
  </group>

  <!-- not use gazebo -->
  <group unless="$(arg enable_gazebo)">
    <!-- load hera description -->
    <include file="$(find hera_description)/launch/load_description.launch">
      <arg name="robot_model" default="$(arg robot_model)"/>
      <arg name="use_jsp_gui" default="true"/>
    </include>
  </group>

  <!-- keyboard teleop -->
  <group if="$(arg enable_teleop)" >
    <node name="teleop_twist_keyboard" pkg="teleop_twist_keyboard" type="teleop_twist_keyboard.py"
      output="screen" launch-prefix="xterm -hold -e"/>
  </group>

  <!-- check collision -->
  <group if="$(arg enable_check_collision)">
    <node name="hera_check_collision" pkg="hera_description" type="hera_check_collision" output="screen">
      <param name="world_name" value="default"/>
      <param name="robot_name" value="$(arg robot_name)"/>
    </node>
  </group>

  <!-- spawn objects to the world -->
  <group if="$(arg spawn_objects)">
    <node pkg="tmc_wrs_gazebo_worlds" type="spawn_objects" name="spawn_objects" args="--seed $(arg seed) --percategory $(arg per_category) --obstacles $(arg obstacles) --perrow $(arg per_row)" output="screen" />
  </group> 
  <!-- task evaluator -->
  <group if="$(arg task_evaluator)">
    <node pkg="tmc_gazebo_task_evaluators" type="setup_score_widget" name="setup_score_widget"/>
    
    <include file="$(find tmc_gazebo_task_evaluators)/launch/robocup2021.launch">
      <arg name="robot_name" value="$(arg robot_name)"/>
      <arg name="camera_controller" value="$(arg camera_controller)"/>
    </include>
  </group>

</launch>

